src/components/Address/withCopy.tsx

```diff
@@ -0,0 +1,73 @@
+import { ComponentType, FC, useEffect, useState } from 'react'
+import { CopySvg } from '@/components/CopySvg'
+import { ADDRESS_ANIMATION_DURATION } from '@/lib/constants'
+import { shortAddress } from '../../lib/utils'
+import { Address, isAddress } from 'viem'
+
+type WithCopyProps = {
+  clipboard: string
+}
+
+type CopyStatus = 'success' | 'error'
+
+export const withCopy = <P extends {}>(Component: ComponentType<P>): FC<P & WithCopyProps> => {
```

Comment:
I noticed @antomor did WithBuilderButton with upper camel case, so I guess we should decide and be consistent. I prefer yours to be honest, as it is an HOC and not used as a component in its own rights.

Reply:
Let's wait what he has to say, and after that Ill do the necessary changes.

Reply:
Sure, let's use the lowercase